Updated README to reflect 3DS 2.0 flow (#167)
braintree/braintree-ios-drop-in: README.md

### 3D-Secure + Drop-In

> The new Drop-In supports 3D-Secure verification. If you have enabled 3D-Secure in the control panel, enable it in the BTDropInRequest and set an amount. Then, create a BTThreeDSecureRequest() object, setting as many fields on it as possible; the more fields that are set, the less likely it is that a user will be be presented with a challenge. Make sure to attach this object to the BTDropInRequest before use.

Reviewer: This is super minor, but I don't think we need the parens on BTThreeDSecureRequest here.

```suggestion
The new Drop-In supports 3D-Secure verification. If you have enabled 3D-Secure in the control panel, enable it in the BTDropInRequest and set an amount. Then, create a BTThreeDSecureRequest object, setting as many fields on it as possible; the more fields that are set, the less likely it is that a user will be be presented with a challenge. Make sure to attach this object to the BTDropInRequest before use.
```

Reviewer 2: Can we put them in backticks?

Author: Done!

> The new Drop-In supports 3D-Secure verification. If you have enabled 3D-Secure in the control panel, enable it in the `BTDropInRequest` and set an amount. Then, create a `BTThreeDSecureRequest object`, setting as many fields on it as possible; the more fields that are set, the less likely it is that a user will be be presented with a challenge. Make sure to attach this object to the `BTDropInRequest` before use.

Reviewer: The backticks should be around `BTThreeDSecureRequest`, not `BTThreeDSecureRequest object`, no?

```suggestion
The new Drop-In supports 3D-Secure verification. If you have enabled 3D-Secure in the control panel, enable it in the `BTDropInRequest` and set an amount. Then, create a `BTThreeDSecureRequest` object, setting as many fields on it as possible; the more fields that are set, the less likely it is that a user will be be presented with a challenge. Make sure to attach this object to the `BTDropInRequest` before use.
```

```swift
let request = BTDropInRequest()

> request.threeDSecureVerification = true

Reviewer 3: This change should be reverted, and the `BTDropInRequest.amount` should definitely not be deprecated. The majority of merchants likely aren't using 3DS, and the `amount` value here is what most of them will be setting.

Author: I agree that it shouldn't be marked deprecated, but this code snippet does live in the 3DS-specific section of the README, so it should probably use the most up-to-date 3DS conventions, no?

Reviewer 3: Never mind, I didn't realize this value was only used for 3DS. It can probably be removed from the doc's then.

Reviewer 2: Is the amount used for Apple Pay as well as 3ds?

Author: ^ @Epreuve looked into this yesterday and couldn't find a conclusive answer; I think since the line isn't directly related to using 3DS 2.0, it's probably more important to get this merged than worry now (and we can always update it later).

### 3D-Secure + Drop-In

> The new Drop-In supports 3D-Secure verification. If you have enabled 3D-Secure in the control panel, then just enable it in the BTDropInRequest and set an amount. Then, create a BTThreeDSecureRequest() object and set as many fields on it as possible; the more fields that are set, the less likely it is for a user to be presented with a challenge. Make sure to attach this object to the BTDropInRequest before use.

Reviewer 4: ```suggestion
The new Drop-In supports 3D-Secure verification. If you have enabled 3D-Secure in the control panel, enable it in the BTDropInRequest, too, and set an amount. Then, create a BTThreeDSecureRequest() object, setting as many additionalInformation fields as possible; the more fields that are set, the less likely you will encounter a decline or the user will be presented with a challenge. Make sure to attach this object to the BTDropInRequest before use.
```

Reviewer 4: This language feels a little too casual, I made some suggestions but I'm not sure if I'm 100% right mentioning `additionalInfo` please double check for accuracy.

> ```swift

Reviewer 2: Should there be an obj c version of this as well?

Author: It's the same process conceptually for both languages; do you mean adding ObjC Code snippets? The whole README is in Swift as of now.

Reviewer: My guess is that anyone who uses Obj C will be able to translate the Swift snippets themselves. So I'd say we're good with just having Swift in the README.

let request = BTDropInRequest()
request.threeDSecureVerification = true

> request.amount = "1.00"

Reviewer: So, I'm always confused about `request.amount` vs. `threeDSecureRequest.amount`, and I have a feeling merchants will be too. My understanding is that we want merchants to use `threeDSecureRequest.amount`, not `request.amount`. So, maybe we should just remove `request.amount` from this code snippet?

I know it's not part of this PR, but shouldn't we also mark `request.amount` as deprecated?

Reviewer 2: ^ That's what I did in the web version of this, so yeah, I think we should do that too to be consistent.

Reviewer: I'll open a separate PR to deprecate `request.amount`.

let threeDSecureRequest = BTThreeDSecureRequest()
threeDSecureRequest.threeDSecureRequestDelegate = self

> threeDSecureRequest.amount = 1.0

Reviewer 4: The amount doesn't need to be two 0's for decimal places?

Author: It's just a float, so that shouldn't matter. We could change the README to 1.00 so it's clearer if you'd prefer, but it's functionally equivalent.

Reviewer 2: Shouldn't this be a string when it's sent to the GW?

Author: From what I could find in the docs and the demo app, it looks like BTDropInRequest.amount is a string and BTThreeDSecureRequest.amount is an NSDecimalNumber.

Reviewer 5: [Confirming](https://github.braintreeps.com/team-sdk/braintree-ios/blob/214c01b8db5fcb965f59ff3f7be56f1536ecf2ef/Demo/Features/3D%20Secure/BraintreeDemoThreeDSecurePaymentFlowViewController.m#L101) it is a `NSDecimalNumber`. Should could write something like ...

``` 
request.amount = [NSDecimalNumber decimalNumberWithString:@"10.32"];
``` 
to be the most clear of casting a string to a number.

Author: We could write `request.amount = NSDecimalNumber(string: "10.00")`, but I feel like that might be even more confusing than just changing this to 10.00.

Reviewer 5: I don't feel too strongly either way. But whichever we use, number or string, should show two decimal places.

Reviewer 2: Definitely agree with @scannillo

Author: Updated! Would love 👀so we can get this merged.

Reviewer: So, I'm always confused about `request.amount` vs. `threeDSecureRequest.amount`, and I have a feeling merchants will be too. My understanding is that we want merchants to use `threeDSecureRequest.amount`, not `request.amount`. So, maybe we should just remove `request.amount` from this code snippet?

I know it's not part of this PR, but shouldn't we also mark `request.amount` as deprecated?

threeDSecureRequest.email = "test@example.com"

> threeDSecureRequest.requested = .version2

Reviewer 4: just doublechecking because new to iOS -  there should be a `.` in `.version2`

Author: Yep, that's Swift shorthand for the enum value `BTThreeDSecureVersion2`.

let address = BTThreeDSecurePostalAddress()
address.givenName = "Jill"
address.surname = "Doe"
address.phoneNumber = "5551234567"
address.streetAddress = "555 Smith St"
address.extendedAddress = "#2"
address.locality = "Chicago"
address.region = "IL"
address.postalCode = "12345"
address.countryCodeAlpha2 = "US"

> request.billingAddress = address

Reviewer 5: I think we should be setting this `billingAddress` on the `threeDSecureRequest` not the `request`.

Reviewer 2: Yeah, that's how we do it on web.

Author: Right, my bad. In the docs, we call the `threeDSecureRequest` `request`, and something must've gotten caught in the copypasta.
